### PR TITLE
syncer: update dependency to avoid exit when meet unsupported binlog

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close https://github.com/pingcap/dm/issues/1682

### What is changed and how it works?
add workaround for the error caused by go-mysql

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: syncer/streamer_controller.go [hunk 1]
```diff
@@ -249,6 +250,9 @@ func (c *StreamerController) GetEvent(tctx *tcontext.Context) (event *replicatio
 
 	event, err = streamer.GetEvent(ctx)
 	cancel()
+	failpoint.Inject("GetEventError", func(_ failpoint.Value) {
```
Comment (lichunzhu) [inline syncer/streamer_controller.go:253]:
```suggestion
	failpoint.Inject("GetEventError", func() {
```

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

**Review by lichunzhu — COMMENTED** [review]
/lgtm

**Review by GMHDBJD — APPROVED** [review]
/lgtm

**Pull request merged**